lib/galaxy/managers/tags.py

```diff
@@ -0,0 +1,58 @@
+from typing import List, Optional
+
+from pydantic import (
+    BaseModel,
+    Field,
+)
+
+from galaxy.app import StructuredApp
+from galaxy.webapps.base.controller import UsesTagsMixin
+from galaxy.webapps.base.webapp import GalaxyWebTransaction
```

Comment:
We really don't want webapps imports anywhere in app. This messes up the package dependency structure I'm trying to establish in ``packages/``. There are some slides on this here https://training.galaxyproject.org/training-material/topics/dev/tutorials/architecture/slides.html#61. 

Pretty much everything in from galaxy.webapps.base.controller predates the managers and the purpose of the managers was to replace those - so importing them here is problematic.

Reply:
Unfortunately that mixin is doing a lot of work still it looks like:

```
ack UsesTagsMixin lib
lib/galaxy/webapps/galaxy/api/item_tags.py
9:    UsesTagsMixin
16:class BaseItemTagsController(BaseGalaxyAPIController, UsesTagsMixin):

lib/galaxy/webapps/galaxy/api/tags.py
8:from galaxy.webapps.base.controller import UsesTagsMixin
15:class TagsController(BaseGalaxyAPIController, UsesTagsMixin):

lib/galaxy/webapps/galaxy/api/users.py
47:    UsesTagsMixin
55:class UserAPIController(BaseGalaxyAPIController, UsesTagsMixin, BaseUIController, UsesFormDefinitionsMixin):

lib/galaxy/webapps/galaxy/api/history_contents.py
35:    UsesTagsMixin
42:class HistoryContentsController(BaseGalaxyAPIController, UsesLibraryMixin, UsesLibraryMixinItems, UsesTagsMixin):

lib/galaxy/webapps/galaxy/controllers/tag.py
11:from galaxy.webapps.base.controller import BaseUIController, UsesTagsMixin
16:class TagsController(BaseUIController, UsesTagsMixin):

lib/galaxy/webapps/base/controller.py
1514:class UsesTagsMixin(SharableItemSecurityMixin):
```

I would suggest a PR to replace it would a manager should precede this PR.

Reply:
Oh, I see, thanks for clarifying it!
I will try to refactor the manager to avoid importing those.

Reply:
Is [this](https://github.com/galaxyproject/galaxy/pull/11224/commits/0eefb616a47cfd63146f0abb07d232599d1560c1) a good compromise? or maybe a bit hacky? 😅